### PR TITLE
Expose LoadFrom Path/NI/Stream APIs from System.Runtime.Loader

### PR DESCRIPTION
**Pull request by gkhanna79**
Fixes https://github.com/dotnet/coreclr/issues/1954

@weshaggard PTAL

### CODE REVIEW

**Comment by gkhanna79** [comment]
Associated with https://github.com/dotnet/coreclr/issues/2618

**Comment by stephentoub** [comment]
Thanks, @gkhanna79.  Are there tests that should be added / updated?  And do we need to update the test-runtime to pull in a newer version of the NuGet package that contains these updates to mscorlib?

Otherwise, the ref visibility changes LGTM.

**Comment by gkhanna79** [comment]
@stephentoub I am not aware of the cadence for test-runtime package updates. When is is typically done? Ideally, we should keep it updated. Can you point me to where the change needs to be made and where do I get the version of CoreCLR package from containing my change?

File: src/System.Runtime.Loader/ref/System.Runtime.Loader.cs [hunk 1]
```diff
@@ -24,10 +24,10 @@ public abstract partial class AssemblyLoadContext
...
-        protected System.Reflection.Assembly LoadFromNativeImagePath(string nativeImagePath, string assemblyPath) { return default(System.Reflection.Assembly); }
-        protected System.Reflection.Assembly LoadFromStream(System.IO.Stream assembly) { return default(System.Reflection.Assembly); }
-        protected System.Reflection.Assembly LoadFromStream(System.IO.Stream assembly, System.IO.Stream assemblySymbols) { return default(System.Reflection.Assembly); }
+        public System.Reflection.Assembly LoadFromAssemblyPath(string assemblyPath) { return default(System.Reflection.Assembly); }
+        public System.Reflection.Assembly LoadFromNativeImagePath(string nativeImagePath, string assemblyPath) { return default(System.Reflection.Assembly); }
+        public System.Reflection.Assembly LoadFromStream(System.IO.Stream assembly) { return default(System.Reflection.Assembly); }
+        public System.Reflection.Assembly LoadFromStream(System.IO.Stream assembly, System.IO.Stream assemblySymbols) { return default(System.Reflection.Assembly); }
         protected System.IntPtr LoadUnmanagedDllFromPath(string unmanagedDllPath) { return default(System.IntPtr); }
```
Comment (davidfowl) [inline src/System.Runtime.Loader/ref/System.Runtime.Loader.cs]: What about this one?

Comment (gkhanna79) [inline src/System.Runtime.Loader/ref/System.Runtime.Loader.cs]: If you need it, let me know and I will expose it. It was not listed in the initial set of APIs requested.

**Comment by gkhanna79** [comment]
The existing tests are sufficient since they validate the same code-path and semantics.

**Comment by stephentoub** [comment]
>  am not aware of the cadence for test-runtime package updates. When is is typically done? Ideally, we should keep it updated. Can you point me to where the change needs to be made and where do I get the version of CoreCLR package from containing my change?

@weshaggard likely has a more comprehensive answer, but basically you update https://github.com/dotnet/corefx/blob/master/src/Common/test-runtime/project.json with the newer version number, which you can get from http://myget.org/gallery/dotnet-core... NuGet packages are published daily, so I expect any changes you checked in to coreclr today should show up in the packages published there tomorrow.

**Comment by gkhanna79** [comment]
Thanks @stephentoub. @weshaggard Let me know if there is anything to add. To confirm, I need to update Microsoft.NetCore.TestHost package version, right?

**Comment by weshaggard** [comment]
You don't actually need to update anything unless something in this repo such as tests wants to consume these changes. I do think we should add some additional test coverage for these however. 

Once we need to consume these APIs we need to do a refresh of all the prerelease packages in our repo following the instructions at https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/project-nuget-dependencies.md#upgrading-a-package-dependency

**Comment by gkhanna79** [comment]
@weshaggard Does this mean I can merge the change tomorrow when CoreCLR is published?

**Comment by weshaggard** [comment]
@dotnet-bot test this please

**Comment by weshaggard** [comment]
Yes please wait until your mscorlib changes are published then merge this when CI is green.

**Pull request merged**